lib/jekyll/renderer.rb

```diff
@@ -33,76 +33,65 @@ def layouts
     # Determine which converters to use based on this document's
     # extension.
     #
-    # Returns an array of Converter instances.
+    # Returns Array of Converter instances.
     def converters
       @converters ||= site.converters.select { |c| c.matches(document.extname) }.sort
     end
 
     # Determine the extname the outputted file should have
     #
-    # Returns the output extname including the leading period.
+    # Returns String the output extname including the leading period.
     def output_ext
       @output_ext ||= (permalink_ext || converter_output_ext)
     end
 
-    ######################
-    ## DAT RENDER THO
-    ######################
-
+    # Prepare payload and render the document
+    #
+    # Returns String rendered document output
     def run
       Jekyll.logger.debug "Rendering:", document.relative_path
 
-      payload["page"] = document.to_liquid
-
-      if document.respond_to? :pager
-        payload["paginator"] = document.pager.to_liquid
-      end
-
-      if document.is_a?(Document) && document.collection.label == "posts"
-        payload["site"]["related_posts"] = document.related_posts
-      else
-        payload["site"]["related_posts"] = nil
-      end
-
-      # render and transform content (this becomes the final content of the object)
-      payload["highlighter_prefix"] = converters.first.highlighter_prefix
-      payload["highlighter_suffix"] = converters.first.highlighter_suffix
+      assign_pages
+      assign_related_posts
+      assign_highlighter_options
 
       Jekyll.logger.debug "Pre-Render Hooks:", document.relative_path
       document.trigger_hooks(:pre_render, payload)
 
+      render_document
+    end
+
+    # Render the document.
+    #
+    # Returns String rendered document output
+    # rubocop: disable AbcSize
+    def render_document
       info = {
+        :filters   => [Jekyll::Filters],
         :registers => { :site => site, :page => payload["page"] }
       }
-
-      output = document.content
-
+      @output = document.content
```

Comment:
Hmm... I think it made more sense before when I changed the signatures.
But since we can't change the signatures, I'll roll it back to just pass it to the methods!